sui/gmp.js

```diff
@@ -0,0 +1,141 @@
+    return '0x' + data.channel.id;
+}
+
+async function sendCommand(keypair, client, contracts, args, options) {
+    const [destinationChain, destinationAddress, feeAmount, payload] = args;
+    const params = options.params;
+
+    const [testConfig, gasServiceConfig] = contracts;
+    const gasServicePackageId = gasServiceConfig.address;
+    const singletonObjectId = testConfig.objects.singleton;
+
+    const unitAmount = getUnitAmount(feeAmount);
+    const walletAddress = keypair.toSuiAddress();
+    const refundAddress = options.refundAddress || walletAddress;
+
+    const channelId = await getChannelId(client, singletonObjectId);
+
+    const tx = new TransactionBlock();
+    const [coin] = tx.splitCoins(tx.gas, [unitAmount]);
+
+    tx.moveCall({
+        target: `${testConfig.address}::test::send_call`,
+        arguments: [
+            tx.object(singletonObjectId),
+            tx.pure(bcs.string().serialize(destinationChain).toBytes()),
+            tx.pure(bcs.string().serialize(destinationAddress).toBytes()),
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(payload)).toBytes()),
+        ],
+    });
+
+    tx.moveCall({
+        target: `${gasServicePackageId}::gas_service::pay_gas`,
+        arguments: [
+            tx.object(gasServiceConfig.objects.GasService),
+            coin, // Coin<SUI>
+            tx.pure.address(channelId), // Channel address
+            tx.pure(bcs.string().serialize(destinationChain).toBytes()), // Destination chain
+            tx.pure(bcs.string().serialize(destinationAddress).toBytes()), // Destination address
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(payload)).toBytes()), // Payload
+            tx.pure.address(refundAddress), // Refund address
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(params)).toBytes()), // Params
+        ],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Call sent', receipt.digest);
+}
+
+async function execute(keypair, client, contracts, args, options) {
+    const [testConfig] = contracts;
+
+    const [sourceChain, messageId, sourceAddress, payload] = args;
+
+    const singletonObjectId = testConfig.objects.singleton;
+    const channelId = await getChannelId(client, singletonObjectId);
+
+    const encodedMessage = approvedMessageStruct
+        .serialize({
+            source_chain: sourceChain,
+            message_id: messageId,
+            source_address: sourceAddress,
+            destination_id: channelId,
+            payload: arrayify(payload),
+        })
+        .toBytes();
+
+    const tx = new TransactionBlock();
+    tx.moveCall({
+        target: `${testConfig.address}::test::execute`,
+        arguments: [tx.pure(bcs.vector(bcs.u8()).serialize(encodedMessage).toBytes()), tx.object(singletonObjectId)],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Call executed', receipt.digest);
+}
```

Comment:
need help here. anyone knows how to send the first argument correctly to call [this function](https://github.com/axelarnetwork/axelar-cgp-sui/blob/74412184f3c40203299aeb3a27773d4713760268/move/test/sources/test/test.move#L61-L70)?